posthog/api/feature_flag.py

```diff
@@ -714,7 +715,31 @@ def _filter_request(self, request: request.Request, queryset: QuerySet) -> Query
 
         for key in filters:
             if key == "active":
-                queryset = queryset.filter(active=filters[key] == "true")
+                if filters[key] == "STALE":
+                    # Get flags that are at least 30 days old and active
+                    # This is an approximation - the serializer will compute the exact status
+                    queryset = queryset.filter(active=True, created_at__lt=thirty_days_ago()).extra(
+                        where=[
+                            """
+                            (
+                                (
+                                    filters->'groups'->0->>'rollout_percentage' = '100'
+                                    AND (filters->'groups'->0->'properties')::text = '[]'::text
+                                    AND (filters->'multivariate' IS NULL OR jsonb_array_length(filters->'multivariate'->'variants') = 0)
+                                )
+                                OR
+                                (
+                                    jsonb_array_length(filters->'multivariate'->'variants') = 1
+                                    AND filters->'multivariate'->'variants'->0->>'rollout_percentage' = '100'
+                                    AND filters->'groups'->0->>'rollout_percentage' = '100'
+                                    AND (filters->'groups'->0->'properties')::text = '[]'::text
+                                )
```

Comment:
I think we should remove this condition (and associated test). Based on our current definition of stale, this might not be true. For example, a variant being 100% could be how someone starts rolling out a feature. And over time, they start changing a different variant to more:

```
day 1: variant-old: 100%, variant-new: 0% (prepping for a new feature)
day 32: (looks stale, but it's not actually stale because we plan to change it)
day 33: variant-old: 90%, variant-new: 10%
day 34: variant-old: 50%, variant-new: 50%
day 35: variant-old: 0%, variant-new: 100%
day 66: Only now is it stale
```

We can't know which variant being rolled out to 100% represents the final state.

Reply:
Fair, on it.

Reply:
@chaitanyya would you like me to take this over or do you still plan to finish it soon? Either way, happy to help.

Reply:
On it now, give me an hour. Sorry about the delay

Reply:
No problem at all!

Reply:
> Also, maybe out of context info, but i opened two more PRs sometime back, would be great if you can review these whenever you get time:

I'll see if I can find the right people to review those changes. If I can't find people in a reasonable time, I'll review them. I'm just not as familiar with that code.

Reply:
Great! appreciate that

Reply:
@haacked just so i understand this correctly, do we also need to change the existing definition for STALE in 

https://github.com/PostHog/posthog/blob/ebec99d7e1bcfc079dbd1cb66a7b125552f31f3c/posthog/models/feature_flag/flag_status.py#L58-L61

Because that file does include features that are fully rolled out as `STALE`. Also, for group and multivariate we do check `rollout_percentage == 100 and len(properties) == 0`

Reply:
Oh! I didn't realie that! In that case, I think what you have is good.

Reply:
Yes, when i was implementing it, i wanted it to be in-line with how we show the "stale" badge on the feature flag items.

I will quickly do a second pass at it and make sure everything is working as expected.

Reply:
Included super_groups and holdout_groups as well same as our flag_status config.

<img width="1111" height="293" alt="Screenshot 2025-07-11 at 4 20 44 PM" src="https://github.com/user-attachments/assets/380fc88e-4e72-4f2f-a652-71e801eadd98" />

Included tests as well and they are passing.